script/sheet/armor.js

```diff
@@ -4,6 +4,28 @@ export class ForbiddenLandsArmorSheet extends ForbiddenLandsItemSheet {
   static get defaultOptions() {
     return mergeObject(super.defaultOptions, {
       template: "systems/forbidden-lands/model/armor.html",
+      tabs: [
+        {
+          navSelector: ".sheet-tabs",
+          contentSelector: ".sheet-body",
+          initial: "attributes",
+        },
+      ],
     });
   }
+
+  computeQuality(data) {
+    data.artifact = data.data.artifactBonus !== "";
```

Comment:
I think I've spotted a small glitch here.
Whenever I create an armor, the "artifact" tab is displayed even if the artifact dropdown tells me "Normal item". I went here and noticed that `computeQuality` returns `true` until you set an artifact die and back to normal item.

This is because the `armor` template does not have an `artifactBonus` field (`undefined` is not an empty string). The same is true for the `gear` template (with the same effect).

I believe that both `weapon`, `gear` and `armor` should include an `artifact` template, the same we do for `craftable`. It would contain everything that is tied to the artifact tab:

- the bonus die
- effect
- drawback
- appearance
- description (perhaps this field should not be artifact-specific… but we can change it later)

Plus, of course, the migration that goes with this change.

Reply:
Yeah I agree. We should actually template this. I was hoping not to have to touch migration but it seems it is necessary.

Reply:
I suggest we add `artifactBonus` to the `bonus` template. I think it makes sense to bunch them together.

(And while migration is of course the correct thing to do here, I could see us cheating in this case and just make the code treat an `undefined` value the same as `""`... 😇 )

Reply:
Yeah, that's a better solution imo. 

One reason I do not necessarily want to touch the template or migration at the moment is due to the news of changes in 0.8.0 It seems we'll be doing a massive amount of migration when 0.8 drops. To the point where it's actually a better idea to restructure the system then while we do. Until then any cheap tricks to avoid doing too much migration are worth looking into. The main reason being this makes the system somewhat more robust should migration to 0.8 break something.

Reply:
Okay! But to be clear, I think we should still update `template.json` to reflect that more items than weapons has the `artifactBonus`. The migration I'm OK with skipping if we make sure to handle existing items where the property is missing.

Reply:
> Okay! But to be clear, I think we should still update `template.json` to reflect that more items than weapons has the `artifactBonus`. The migration I'm OK with skipping if we make sure to handle existing items where the property is missing.

Yes, yes. This is an oversight I should have added it to the template. 

For existing items we can instead check for regex /d8|d10|d12/i perhaps? Would avoid artifact being toggled unless its a valid artifact.

Reply:
I would just go with the simple `data.artifact = !!data.data.artifactBonus;`, since both `""` and `undefined` are both falsy values. I expect if the property was somehow set to `null` or any other falsy value we would still want to treat it that way.

Reply:
Also okay to skip the migration for now since we have decent workarounds within our reach.

@aMediocreDad Not exactly on topic, but can you share more details about those impacting changes coming with Foundry 0.8? A link should do, I just want to understand what motivates your decision and also understand how we might anticipate those changes. Thanks! 😄

Reply:
@giant-teapot https://foundryvtt.wiki/en/migrations/0_8_0

There'll also be a stream by Andrew tonight at 2pm EST: https://twitch.tv/foundryvtt